client/windows/settings/SettingsMainContainer.cpp

```diff
@@ -0,0 +1,147 @@
+/*
+ * SettingsMainContainer.cpp, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+
+#include <SDL_events.h>
+#include "StdInc.h"
+
+#include "SettingsMainContainer.h"
+
+#include "GeneralOptionsTab.h"
+#include "AdventureOptionsTab.h"
+#include "BattleOptionsTab.h"
+#include "OtherOptionsWindow.h"
+
+#include "filesystem/ResourceID.h"
+#include "CGeneralTextHandler.h"
+#include "gui/CGuiHandler.h"
+#include "lobby/CSavingScreen.h"
+#include "widgets/Images.h"
+#include "CGameInfo.h"
+#include "CPlayerInterface.h"
+#include "CServerHandler.h"
+
+
+SettingsMainContainer::SettingsMainContainer(BattleInterface * parentBattleUi) : InterfaceObjectConfigurable()
+{
+	OBJ_CONSTRUCTION_CAPTURING_ALL_NO_DISPOSE;
+
+	const JsonNode config(ResourceID("config/widgets/settings/settingsMainContainer.json"));
+	addCallback("activateSettingsTab", [this](int tabId) { openTab(tabId); });
+	addCallback("loadGame", [this](int) { loadGameButtonCallback(); });
+	addCallback("saveGame", [this](int) { saveGameButtonCallback(); });
+	addCallback("restartGame", [this](int) { restartGameButtonCallback(); });
+	addCallback("quitGame", [this](int) { quitGameButtonCallback(); });
+	addCallback("returnToMainMenu", [this](int) { mainMenuButtonCallback(); });
+	addCallback("closeWindow", [this](int) { backButtonCallback(); });
+	build(config);
+
+	std::shared_ptr<CPicture> background = widget<CPicture>("background");
+	pos.w = background->pos.w;
+	pos.h = background->pos.h;
+	pos = center();
+
+	std::shared_ptr<CButton> loadButton = widget<CButton>("loadButton");
+	assert(loadButton);
+
+	std::shared_ptr<CButton> saveButton = widget<CButton>("saveButton");
+	assert(saveButton);
+
+	std::shared_ptr<CButton> restartButton = widget<CButton>("restartButton");
+	assert(restartButton);
+
+	if(CSH->isGuest())
+	{
+		loadButton->block(true);
+		saveButton->block(true);
+		restartButton->block(true);
+	}
+
+	int defaultTabIndex = 0;
+	if(settings["general"]["lastSettingsTab"].isNumber())
+		defaultTabIndex = settings["general"]["lastSettingsTab"].Integer();
```

Comment:
I think that if window is opened from battle we should unconditionally open battle tab instead

Reply:
done